File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/GroupFragAdm/DiscussionPanelFragAdm/adapterRecVDisPanelMainSc.java
package com.example.adminapp.GroupFragAdm.DiscussionPanelFragAdm;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.adminapp.R;
import com.google.android.material.circularreveal.cardview.CircularRevealCardView;

import java.util.ArrayList;

public class adapterRecVDisPanelMainSc extends RecyclerView.Adapter<adapterRecVDisPanelMainSc.myviewholder> {

    private ArrayList<DiscussionPanelData> dataHolder;
    private ArrayList<DiscussionPanelData> dataHolderBackup;

    public adapterRecVDisPanelMainSc(ArrayList<DiscussionPanelData> dataHolder) {
        this.dataHolder = dataHolder;
        dataHolderBackup = new ArrayList<>(dataHolder);
    }

    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new myviewholder(LayoutInflater.from(parent.getContext()).inflate(R.layout.single_row_des_recv_dispanel_mainsc_fadm,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull myviewholder holder, int position) {
        holder.titleTvSingleRDDisPanelMainScFAdm.setText(dataHolder.get(position).getTitle());
        holder.descripTvSingleRDDisPanelMainScFAdm.setText(dataHolder.get(position).getDescription());
    }

    @Override
    public int getItemCount() {
        return dataHolder.size();
    }

    public class myviewholder extends RecyclerView.ViewHolder {
        CircularRevealCardView circularCardVSingleRDDisPanelMainScFAdm;
        TextView titleTvSingleRDDisPanelMainScFAdm,descripTvSingleRDDisPanelMainScFAdm;
        public myviewholder(@NonNull View itemView) {
            super(itemView);

            circularCardVSingleRDDisPanelMainScFAdm = itemView.findViewById(R.id.circularCardVSingleRDDisPanelMainScFAdm);
            titleTvSingleRDDisPanelMainScFAdm = itemView.findViewById(R.id.titleTvSingleRDDisPanelMainScFAdm);
            descripTvSingleRDDisPanelMainScFAdm = itemView.findViewById(R.id.descripTvSingleRDDisPanelMainScFAdm);
        }
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/GroupFragAdm/DiscussionPanelFragAdm/adapterRecVDisPanelMainSc.kt
package com.example.adminapp.GroupFragAdm.DiscussionPanelFragAdm

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.adminapp.R
import com.google.android.material.circularreveal.cardview.CircularRevealCardView

class adapterRecVDisPanelMainSc(private val dataHolder: ArrayList<DiscussionPanelData>?) :
    RecyclerView.Adapter<adapterRecVDisPanelMainSc.myviewholder>() {
    private val dataHolderBackup: ArrayList<DiscussionPanelData>

    init {
        dataHolderBackup = ArrayList(dataHolder)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): myviewholder {
        return myviewholder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.single_row_des_recv_dispanel_mainsc_fadm, parent, false)
        )
    }

    override fun onBindViewHolder(holder: myviewholder, position: Int) {
        holder.titleTvSingleRDDisPanelMainScFAdm.text = dataHolder!![position].title
        holder.descripTvSingleRDDisPanelMainScFAdm.text = dataHolder[position].description
    }

    override fun getItemCount(): Int {
        return dataHolder!!.size
    }

    inner class myviewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var circularCardVSingleRDDisPanelMainScFAdm: CircularRevealCardView
        var titleTvSingleRDDisPanelMainScFAdm: TextView
        var descripTvSingleRDDisPanelMainScFAdm: TextView

        init {
            circularCardVSingleRDDisPanelMainScFAdm =
                itemView.findViewById(R.id.circularCardVSingleRDDisPanelMainScFAdm)
            titleTvSingleRDDisPanelMainScFAdm =
                itemView.findViewById(R.id.titleTvSingleRDDisPanelMainScFAdm)
            descripTvSingleRDDisPanelMainScFAdm =
                itemView.findViewById(R.id.descripTvSingleRDDisPanelMainScFAdm)
        }
    }
}
